<launch>

    <!-- This node will launch frame conversion from vision pose (tf) to mavros pose -->
    <arg name="target_frame_id"   default="camera_init" />
    <param name="target_frame_id"   value="$(arg target_frame_id)" />

    <arg name="source_frame_id"   default="body" />
    <param name="source_frame_id"   value="$(arg source_frame_id)" />

    <param name="output_rate"       value="10" />

    <!-- 自动误差修正延时，默认节点启动后2s采集一次误差作为全局误差，设为0.0来禁用误差修正 -->
    <arg name="auto_offset_delay" default="2.0" />
    <param name="auto_offset_delay" value="$(arg auto_offset_delay)" />

    <!-- 添加旋转参数 -->
    <arg name="roll_lidar" default="0.0" />
    <param name="roll_lidar" value="$(arg roll_lidar)" />

    <arg name="pitch_lidar" default="0.0" />
    <param name="pitch_lidar" value="$(arg pitch_lidar)" />

    <arg name="yaw_lidar" default="0.0" />
    <param name="yaw_lidar" value="$(arg yaw_lidar)" />

    <!-- 添加平移参数 -->
    <arg name="x_lidar" default="0.0" />
    <param name="x_lidar" value="$(arg x_lidar)" />

    <arg name="y_lidar" default="0.0" />
    <param name="y_lidar" value="$(arg y_lidar)" />

    <arg name="z_lidar" default="0.0" />
    <param name="z_lidar" value="$(arg z_lidar)" />

    <!-- 添加全局旋转参数 -->
    <arg name="gamma_world" default="-1.5707963" />
    <param name="gamma_world" value="$(arg gamma_world)" />

    <node pkg="vision_to_mavros" type="mid360_to_mavros_node" name="mid360_to_mavros" output="screen" >
        <remap from="mid360_pose" to="/mavros/vision_pose/pose" />
    </node>
</launch>
